der/src/byte_vec.rs

```diff
@@ -0,0 +1,106 @@
+//! Common handling for types backed by byte slices with enforcement of a
+//! library-level length limitation i.e. `Length::max()`.
+
+use crate::{
+    str_slice::StrSlice, DecodeValue, DerOrd, EncodeValue, Error, Header, Length, Reader, Result,
+    Writer,
+};
+use alloc::vec::Vec;
+use core::cmp::Ordering;
+
+/// Byte slice newtype which respects the `Length::max()` limit.
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub(crate) struct ByteVec {
+    /// Precomputed `Length` (avoids possible panicking conversions)
+    length: Length,
+
+    /// Inner value
+    inner: Vec<u8>,
+}
```

Comment:
I think having a type for representing an owned bytestring is a good idea, but I'm wondering if it should internally represent that bytestring with a `Box<[u8]>` and perhaps be called simply `Bytes`.

I think `Box<[u8]>` is a better choice for representing immutable data (i.e. this type presents an immutable API) and is also slightly more compact.

Reply:
Good point